src/app/file/tiff_format.cpp

```diff
@@ -0,0 +1,179 @@
+// Aseprite
+// Copyright (c) 2018-2020  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+//
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "app/console.h"
+#include "app/context.h"
+#include "app/doc.h"
+#include "app/file/file.h"
+#include "app/file/file_format.h"
+#include "app/file/format_options.h"
+#include "app/pref/preferences.h"
+#include "base/cfile.h"
+#include "base/clamp.h"
+#include "base/file_handle.h"
+#include "doc/doc.h"
+#include "ui/window.h"
+
+#include "tinytiff_defs.h"
+#include "tinytiff_export.h"
+#include "tinytiff_tools.hxx"
+#include "tinytiffreader.hxx"
+#include "tinytiffwriter.h"
+
+
+namespace app {
+
+using namespace base;
+
+class TiffFormat : public FileFormat {
+
+  const char* onGetName() const override {
+    return "tiff";
+  }
+
+  void onGetExtensions(base::paths& exts) const override {
+    exts.push_back("tiff");
+    exts.push_back("tif");
+  }
+
+  dio::FileFormat onGetDioFormat() const override {
+    return dio::FileFormat::TIFF_IMAGE;
+  }
+
+  int onGetFlags() const override {
+    return
+      FILE_SUPPORT_LOAD |
+      FILE_SUPPORT_RGB |
+      FILE_SUPPORT_RGBA |
+      FILE_SUPPORT_GRAY |
+      FILE_SUPPORT_GRAYA |
+      FILE_SUPPORT_INDEXED |
+      FILE_SUPPORT_SEQUENCES |
+      FILE_SUPPORT_GET_FORMAT_OPTIONS |
+      FILE_SUPPORT_PALETTE_WITH_ALPHA;
+  }
+
+  bool onLoad(FileOp* fop) override;
+#ifdef ENABLE_SAVE
+  bool onSave(FileOp* fop) override;
+#endif
+};
+
+FileFormat* CreateTiffFormat()
+{
+  return new TiffFormat;
+}
+
+bool TiffFormat::onLoad(FileOp* fop)
+{
+  PixelFormat pixelFormat;
+  int imageWidth = 100;
+  int imageHeight = 100;
+
+  //FileHandle handle(open_file_with_exception(fop->filename(), "rb"));
+  //FILE* file = handle.get();
+
+  TinyTIFFReaderFile* tiffr = NULL;
+  tiffr = TinyTIFFReader_open(fop->filename().c_str());
```

Comment:
Does this need to be initialized to NULL? You're setting it to its real value on the very next line.
```suggestion
  TinyTIFFReaderFile* tiffr = TinyTIFFReader_open(fop->filename().c_str());
```